packages/core-plugin-api/src/extensions/extensions.tsx

```diff
@@ -237,17 +238,23 @@ export function createReactExtension<
 
         return (
           <Suspense fallback={<Progress />}>
-            <PluginErrorBoundary app={app} plugin={plugin}>
-              <AnalyticsContext
-                attributes={{
-                  pluginId: plugin.getId(),
-                  ...(name && { extension: name }),
-                  ...(mountPoint && { routeRef: mountPoint.id }),
-                }}
-              >
-                <Component {...props} />
-              </AnalyticsContext>
-            </PluginErrorBoundary>
+            <PluginContextProvider
+              componentName={componentName}
+              plugin={plugin}
+              routeRef={mountPoint?.id}
+            >
+              <PluginErrorBoundary app={app} plugin={plugin}>
+                <AnalyticsContext
```

Comment:
Would be interesting to move this `<AnalyticsContext>` into the `<PluginContextProvider>`, especially if it had access to decorated metadata from #10256 (the most useful stuff in there, IMHO, is the plugin owner and the plugin version).

Reply:
The `<AnalyticsContext>` could be a non-viewed empty component `<AnalyticsListener />` which does `usePluginContextStack`, potentially.

After all, the idea (@Rugvip's question above) of an actual _stack_ was taken from the AnalyticsContext actually.

Reply:
+1 to avoiding duplication here. Gotta keep in mind that we'd want to support standalone `<AnalyticsContext />` usage still though right?

Reply:
As I wrote in https://github.com/backstage/backstage/pull/10258#issuecomment-1070903156 about a `CustomContext`, I wonder if it'd make sense to have a component that you can add anywhere to create an explicit boundary with possible intent and arbitrary "value". This could then in turn use the PluginContextProvider itself. That'd end up adding it to the context stack, retrievable by `usePluginContextStack()`. Maybe call it `BackstageBoundary` with props `intent: string; value: any`. The intent for `createReactExtension` could be `"plugin"` e.g.

You could have
```tsx
const tree = <Foo>
  <BackstageBoundary intent="analytics" value={{ some: "value" }}>
    <Bar>
      <BackstageBoundary intent="abtest" value="legacy">
        <Baz />
      </BackstageBoundary>
    </Bar>
  </BackstageBoundary>
</Foo>
```

The `<AnalyticsListener />` could `usePluginContextStack()` and only care about, say, "plugin" and "analytics":

```ts
const stack = usePluginContextStack()
  .filter(boundary =>
    boundary.intent === 'plugin' || boundary.intent === 'analytics'
  );
// As by the tree above ('plugin' being injected outside this tree from the createReactExtension):
// [
//   { intent: 'analytics', pluginId: 'my-plugin', value: { some: 'value' }, ... },
//   { intent: 'plugin', pluginId: 'my-plugin', ... },
// ]
```

It's just an idea, but would be a reusable concept for other kinds of boundaries that plugins can use too. You'd loose typings of the value, but perhaps that can be worked out.